code/DynamicCache.php

```diff
@@ -384,6 +393,10 @@ public function run($url)
         // This prevents a new user's security token from being regenerated incorrectly
         $_SESSION['SecurityID'] = SecurityToken::getSecurityID();
 
+        // Set the stage of the website
+        // This is normally called in VersionedRequestFilter.
+        Versioned::choose_site_stage();
```

Comment:
I'm assuming this works because at the start of Versioned::choose_site_stage(), it does this:
```
$preexistingMode = $session
			? $session->inst_get('readingMode')
			: Session::get('readingMode');
```